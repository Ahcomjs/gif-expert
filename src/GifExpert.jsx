import { useState } from "react"
import { AddCategory } from "./components/AddCategory";
import logo from '../src/assets/img/gif-file.png';
import { GifGrid } from './components/GifGrid';


export const GifExpert = () => {

    const [categories, setCategories] = useState([]);

    // const capitalizeLetters = (category) => {
    //     const firstLetter = category.substring(0, 1).toUpperCase();
    //     const restOfLetters = category.slice(1, category.length);
    //     return firstLetter + restOfLetters;
    // }

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;

        setCategories([...categories, newCategory]);
    }

    return (
        <>
            <div className="container">
                <img src={logo} alt="logo" className="logo" />
                <h1>Gif Expert  <span>APP</span></h1>

                <AddCategory
                    onNewCategory={onAddCategory}

                />

                {
                    categories.map((category) => (
                        <GifGrid
                            key={category}
                            category={category} />
                    ))
                }

            </div>

        </>
    )
}