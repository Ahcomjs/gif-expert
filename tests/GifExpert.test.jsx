import { render, screen } from "@testing-library/react";
import { GifExpert } from "../src/GifExpert";

describe('Test GifExpert', () => {
    test('Should match the Snapshops', () => {
        const { container } = render(<GifExpert />);
        screen.debug();
        expect(container).toMatchSnapshot();
    });
    

})
