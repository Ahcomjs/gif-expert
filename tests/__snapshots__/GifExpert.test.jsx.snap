// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExpert Should match the Snapshops 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="logo"
      class="logo"
      src="test-file-stub"
    />
    <h1>
      Gif Expert 
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Write a name and press enter..."
        type="text"
        value=""
      />
    </form>
    <div
      class="gif-container"
    />
  </div>
</div>
`;
